### PR TITLE
Introduced a minimal reference-rule mapping generator.

### PR DESCRIPTION
**Pull request by matejak**
Generate a simple reference-rules HTML page for each product to `build/tables-<product>-all.html`. References and rules are sorted lexicographically.
This PR can serve as a PoC of a lightweight table generation.

Advantages:

- The code is jinaj2 and Pyhon-based, so it is very small and customizable, and it can be tuned easily (e.g. do version-based sorting for version-like reference IDs)
- As the functionality supports HTML and JSON ordered output, it can serve as a support for modifications to the references processing.

Problems with the current table generation:

- Tables are user-unfriendly - it is not uncommon for one control to be referenced in multiple table rows, which may not be consecutive.
- Introduction of a new reference type requires a lot of XML copy-pasting.

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python file utils/gen_tables.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python file utils/gen_tables.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

File: cmake/SSGCommon.cmake [hunk 1]
```diff
@@ -700,6 +700,18 @@ macro(ssg_make_html_stats_for_product PRODUCT)
     )
 endmacro()
 
+macro(ssg_make_all_tables PRODUCT)
+    add_custom_command(
+        OUTPUT "${CMAKE_BINARY_DIR}/tables-${PRODUCT}-all.html"
+	COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/utils/gen_tables.py" --build-dir "${CMAKE_BINARY_DIR}" --output-type html --output "${CMAKE_BINARY_DIR}/tables-${PRODUCT}-all.html" "${PRODUCT}"
```
Comment (redhatrises) [inline cmake/SSGCommon.cmake:706]:
```suggestion
	COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/utils/gen_tables.py" --build-dir "${CMAKE_BINARY_DIR}" --output-type html --output "${CMAKE_BINARY_DIR}/tables/tables-${PRODUCT}-all.html" "${PRODUCT}"
```

File: cmake/SSGCommon.cmake [hunk 2]
```diff
@@ -700,6 +700,18 @@ macro(ssg_make_html_stats_for_product PRODUCT)
...
+        # Actually we mean that it depends on resolved rules - see also ssg_build_templated_content
+        DEPENDS generate-internal-${PRODUCT}-shorthand.xml
+    )
+    add_custom_target(generate-ssg-tables-${PRODUCT}-all
+        DEPENDS "${CMAKE_BINARY_DIR}/tables-${PRODUCT}-all.html"
```
Comment (redhatrises) [inline cmake/SSGCommon.cmake:711]:
```suggestion
        DEPENDS "${CMAKE_BINARY_DIR}/tables/tables-${PRODUCT}-all.html"
```

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python file utils/gen_tables.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python file utils/gen_tables.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by matejak** [comment]
test this please

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python file utils/gen_tables.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by redhatrises** [comment]
Have no problem with this PR, and the build failure is due to a Fedora communication issue.

**Pull request merged**